types/src/main.ts

```diff
@@ -3372,6 +3372,26 @@ export type CodeActionDisabled = {
 	reason: string;
 };
 
+/**
+ * Code action tags are extra annotations that tweak the rendering of a code action.
+ *
+ * @since 3.18.0 - proposed
+ */
+export namespace CodeActionTag {
+	/**
+	 * Render a code action as LLM-generated, usually using a sparkle icon.
```

Comment:
We should not enforce rendering. I would simply say: Marks the code action as LLM-generated.